src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/DataCenterFormula.kt

```diff
@@ -382,16 +384,18 @@ class DataCenterFormula private constructor(
             loadBalancer.waitUntilHealthy(Duration.ofMinutes(5))
         }
 
-        val jira = Jira(
-            nodes = nodes,
-            jiraHome = RemoteLocation(
-                sharedHomeSsh.host,
-                sharedHome.get().remoteSharedHome
-            ),
-            database = databaseDataLocation,
-            address = loadBalancer.uri,
-            jmxClients = jiraNodes.mapIndexed { i, node -> configs[i].remoteJmx.getClient(node.publicIpAddress) }
+        val jiraHomeLocation = RemoteLocation(
+            sharedHomeSsh.host,
+            sharedHome.get().remoteSharedHome
         )
+        val loadBalancerResultsSource = (provisionedLoadBalancer.loadBalancer as? DiagnosableLoadBalancer)
```

Comment:
First time used this `as?` syntax. Casts are usually bad, however given the circumstances of our current model I think this is a better solution than putting a new field into `ProvisionedLoadBalancer` and I'm also not certain if adding methods to `LoadBalancer` would be a good idea. Extending the interface and detecting it allows for wrapping of `DiagnosableLoadBalancer` implementations without losing the feature.

Reply:
Indeed casts are bad. See https://github.com/atlassian/aws-infrastructure/blob/release-2.29.0/src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/DataCenterFormula.kt#L360:
```kotlin
if (loadBalancer is ApacheProxyLoadBalancer) listOf(loadBalancer::updateJiraConfiguration) else emptyList()
```
Fortunately, `ApacheProxyLoadBalancer` will match both `is`/`as?` casts, so it will function.
We need hooks API to avoid further tight coupling. If the apache logs are necessary, then we can merge the tech debt. But if not, then let's skip that part.